src/borg/locking.py

```diff
@@ -356,21 +356,14 @@ def acquire(self, exclusive=None, remove=None, sleep=None):
     def _wait_for_readers_finishing(self, remove, sleep):
         timer = TimeoutTimer(self.timeout, sleep).start()
         while True:
-            self._lock.acquire()
-            try:
+            with self._lock:
                 if remove is not None:
                     self._roster.modify(remove, REMOVE)
                 if len(self._roster.get(SHARED)) == 0:
                     return  # we are the only one and we keep the lock!
                 # restore the roster state as before (undo the roster change):
                 if remove is not None:
                     self._roster.modify(remove, ADD)
-            except:
-                # avoid orphan lock when an exception happens here, e.g. Ctrl-C!
-                self._lock.release()
-                raise
-            else:
-                self._lock.release()
```

Comment:
... that.

somehow I thought that the above change is equivalent.

but suddenly a lot of test failures!?